src/ligature.rs

```diff
@@ -0,0 +1,146 @@
+use skrifa::{
+    charmap::Charmap,
+    raw::{
+        tables::gsub::{CoverageTable, LigatureMarker, LigatureSet, SubstitutionSubtables},
+        types::Offset16,
+        ArrayOfOffsets, FontRef, ReadError, TableProvider, TableRef,
+    },
+    GlyphId,
+};
+use std::{cell::RefCell, iter};
+
+use crate::error::IconResolutionError;
+
+pub struct Ligature {
+    // Includes all components, not just 1..N.
+    pub components: Vec<GlyphId>,
+    pub glyph: GlyphId,
+}
+
+type LigatureInnerIter<'a> = Vec<(
+    GlyphId, // First gid in ligature components.
+    RefCell<Box<dyn Iterator<Item = Result<TableRef<'a, LigatureMarker>, ReadError>> + 'a>>,
+)>;
+pub struct LigatureIter<'a> {
+    inner_iter: LigatureInnerIter<'a>,
+    // current position in `inner_iter`
+    idx: usize,
+    // 1..N, if present, the iterator will be targeted to return only ligatures matching `gids`
+    gids: Option<Vec<GlyphId>>,
+}
+
+impl<'a> Iterator for LigatureIter<'a> {
+    type Item = Result<Ligature, IconResolutionError>;
```

Comment:
The types look like they are general purpose over all ligatures but then some of the capabilities and error types make them unique to icons. I think the general purpose version is nice.